tests/subsys/logging/log_core/src/log_core_test.c

```diff
@@ -173,11 +173,186 @@ static void log_setup(bool backend2_enable)
 	test_source_id = log_source_id_get(STRINGIFY(LOG_MODULE_NAME));
 }
 
+/*
+ * Test is using 2 backends and runtime filtering is enabled. After first call
+ * filtering for backend2 is reduced to warning. It is expected that next INFO
+ * level log message will be passed only to backend1.
+ */
+static void test_log_backend_runtime_filtering(void)
+{
+	log_setup(true);
+
+	backend1_cb.check_timestamp = true;
+	backend2_cb.check_timestamp = true;
+
+	backend1_cb.exp_timestamps[0] = 0U;
+	backend1_cb.exp_timestamps[1] = 1U;
+	backend1_cb.exp_timestamps[2] = 2U;
+
+	/* Expect one less log message */
+	backend2_cb.exp_timestamps[0] = 0U;
+	backend2_cb.exp_timestamps[1] = 2U;
+
+	LOG_INF("test");
+	while (log_process(false)) {
+	}
+
+	log_filter_set(&backend2,
+			CONFIG_LOG_DOMAIN_ID,
+			test_source_id,
+			LOG_LEVEL_WRN);
+
+	LOG_INF("test");
+	LOG_WRN("test");
+
+	while (log_process(false)) {
+	}
+
+	zassert_equal(3,
+		      backend1_cb.counter,
+		      "Unexpected amount of messages received by the backend.");
+
+	zassert_equal(2,
+		      backend2_cb.counter,
+		      "Unexpected amount of messages received by the backend.");
+}
+
+/*
+ * When LOG_MOVE_OVERFLOW is enabled, logger should discard oldest messages when
+ * there is no room. However, if after discarding all messages there is still no
+ * room then current log is discarded.
+ */
+uint8_t data[CONFIG_LOG_BUFFER_SIZE];
+static void test_log_overflow(void)
+{
+	uint32_t msgs_in_buf = CONFIG_LOG_BUFFER_SIZE/sizeof(union log_msg_chunk);
+	uint32_t max_hexdump_len = LOG_MSG_HEXDUMP_BYTES_HEAD_CHUNK +
+			    HEXDUMP_BYTES_CONT_MSG * (msgs_in_buf - 1);
+	uint32_t hexdump_len = max_hexdump_len - HEXDUMP_BYTES_CONT_MSG;
+
+
+	zassert_true(IS_ENABLED(CONFIG_LOG_MODE_OVERFLOW),
+		     "Test requires that overflow mode is enabled");
+
+	log_setup(false);
+	backend1_cb.check_timestamp = true;
+	backend2_cb.check_timestamp = true;
+
+	/* expect first message to be dropped */
+	backend1_cb.exp_timestamps[0] = 1U;
+	backend1_cb.exp_timestamps[1] = 2U;
+
+	LOG_INF("test");
+	LOG_INF("test");
+	LOG_HEXDUMP_INF(data, hexdump_len, "test");
+
+	while (log_process(false)) {
+	}
+
+	/* Expect big message to be dropped because it does not fit in.
+	 * First message is also dropped in the process of finding free space.
+	 */
+	backend1_cb.exp_timestamps[2] = 3U;
+
+	LOG_INF("test");
+	LOG_HEXDUMP_INF(data, max_hexdump_len+1, "test");
+
+	while (log_process(false)) {
+	}
+
+	zassert_equal(2,
+		      backend1_cb.counter,
+		      "Unexpected amount of messages received by the backend.");
+}
+
+/*
+ * Test checks if arguments are correctly processed by the logger.
+ *
+ * Log messages with supported number of messages are called. Test backend
+ * validates number of arguments and values.
+ */
+static void test_log_arguments(void)
```

Comment:
same/similar tests are already in `log_api` suite. What do we gain by adding those here?

Reply:
Thank you @nordic-krch , I didn't realize that, updated.